fdbserver/workloads/SpecialKeySpaceCorrectness.actor.cpp

```diff
@@ -1072,6 +1074,132 @@ struct SpecialKeySpaceCorrectnessWorkload : TestWorkload {
+					tx->set(SpecialKeySpace::getManagementApiCommandPrefix("advanceversion"), std::to_string(v2));
+					wait(tx->commit());
+					ASSERT(false); // Should fail with commit_unknown_result
+				} catch (Error& e) {
+					TraceEvent(SevDebug, "AdvanceVersionCommitFailure").error(e);
+					wait(tx->onError(e));
+				}
+			}
+			tx->reset();
+		} catch (Error& e) {
+			wait(tx->onError(e));
+		}
+		// profile client get
+		loop {
+			try {
+				tx->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+				// client_txn_sample_rate
+				state Optional<Value> txnSampleRate =
+				    wait(tx->get(LiteralStringRef("client_txn_sample_rate")
+				                     .withPrefix(SpecialKeySpace::getManagementApiCommandPrefix("profile"))));
+				ASSERT(txnSampleRate.present());
+				Optional<Value> txnSampleRateKey = wait(tx->get(fdbClientInfoTxnSampleRate));
+				if (txnSampleRateKey.present()) {
+					const double sampleRateDbl =
+					    BinaryReader::fromStringRef<double>(txnSampleRateKey.get(), Unversioned());
+					if (!std::isinf(sampleRateDbl)) {
+						ASSERT(txnSampleRate.get().toString() == boost::lexical_cast<std::string>(sampleRateDbl));
+					} else {
+						ASSERT(txnSampleRate.get().toString() == "default");
+					}
+				} else {
+					ASSERT(txnSampleRate.get().toString() == "default");
+				}
+				// client_txn_size_limit
+				state Optional<Value> txnSizeLimit =
+				    wait(tx->get(LiteralStringRef("client_txn_size_limit")
+				                     .withPrefix(SpecialKeySpace::getManagementApiCommandPrefix("profile"))));
+				ASSERT(txnSizeLimit.present());
+				Optional<Value> txnSizeLimitKey = wait(tx->get(fdbClientInfoTxnSizeLimit));
+				if (txnSizeLimitKey.present()) {
+					const int64_t sizeLimit =
+					    BinaryReader::fromStringRef<int64_t>(txnSizeLimitKey.get(), Unversioned());
+					if (sizeLimit != -1) {
+						ASSERT(txnSizeLimit.get().toString() == boost::lexical_cast<std::string>(sizeLimit));
+					} else {
+						ASSERT(txnSizeLimit.get().toString() == "default");
+					}
+				} else {
+					ASSERT(txnSizeLimit.get().toString() == "default");
+				}
+				tx->reset();
+				break;
+			} catch (Error& e) {
+				TraceEvent(SevDebug, "ProfileClientGet").error(e);
+				wait(tx->onError(e));
+			}
+		}
+		{
+			state double r_sample_rate = deterministicRandom()->random01();
+			state int64_t r_size_limit = deterministicRandom()->randomInt64(1e3, 1e6);
+			// update the sample rate and size limit
+			loop {
+				try {
+					tx->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
+					tx->set(LiteralStringRef("client_txn_sample_rate")
+					            .withPrefix(SpecialKeySpace::getManagementApiCommandPrefix("profile")),
+					        Value(boost::lexical_cast<std::string>(r_sample_rate)));
+					tx->set(LiteralStringRef("client_txn_size_limit")
+					            .withPrefix(SpecialKeySpace::getManagementApiCommandPrefix("profile")),
+					        Value(boost::lexical_cast<std::string>(r_size_limit)));
```

Comment:
Can we add a test where we set it the value to something invalid? (e.g. not a number or "default")

Reply:
Done. Passed correctness tests.